### DISCUSSION TITLE
Redact functionality

### DISCUSSION THREAD

**Discussion by Sg-23** [vectordotdev/vector#11496] [Q&A] [answered]
I am working on an implementation to redact credit card numbers, but with regex, I am getting false positives. I need to implement luhn checker for the same. Do we have any inbuilt functionality in vector or can the same be achieved with VRL? Will it require lua for the same?

**Comment by jszwedko** [comment, marked as answer]
Hi @Sg-23 !

I think you'll need to use Lua for this for now. At the least, VRL needs a way to index into a string before you could implement the luhn algorithm.
